### DISCUSSION TITLE
Rendering mode changes the values of axes

### DISCUSSION THREAD

**Discussion by FGropp** [pyvista/pyvista#5483] [Q&A] [answered]
I encountered a confusing behavior in pyvista concerning the scaling of the axes, which seems to depend on the rendering mode. In my example I  calculated the points of a molecule and plotted them using pyvista. The x-values ranges from -1.53815 to 3.0763. But in one case the x values are far bigger as 3.0763, However, if I calculate the maximum x values of the structured grid it is (as expected) 3.0763. 
<img width="899" alt="image" src="https://github.com/pyvista/pyvista/assets/78682764/db5d48c7-df88-46da-8ec2-c6fe96223505">

However, by changing the rendering mode, the axis look like expected. 
<img width="761" alt="image" src="https://github.com/pyvista/pyvista/assets/78682764/1a59acfa-edf7-43e5-a166-1e22b222335a">
What is the reason for this?

**Comment by jourdain** [comment]
The axis grid logic is not the same between VTK/C++ and vtk.js.

**Reply by FGropp** [reply]
Thanks for your answer. Do you know the deeper reason why this might lead to such different results? Normally, I would expect the same result, as the underlying grid is the same. Is there any good documentation that highlights the difference between the two logics? Or how is the spacing of the grid is calculated depending on the grid points?

**Reply by FGropp** [reply]
@banesullivan Maybe you could help?

**Reply by banesullivan** [reply, marked as answer]
When we serialize the scene to send to the client browser fro rendering, the only facet of the axes grid that is serialized is just the fact that it is enabled. We do not actually send the geometrries, text, etc that make uyp the grid. Then vtk.js uses its own mechanisms for creating a grid around the data in the scene which we have little to no ability to customize.

HTH

**Reply by FGropp** [reply]
thank you!
